### PR TITLE
Updated Developer Guide

### PR DESCRIPTION
**Pull request by Benjykoh**
Improved on some formatting issues.

Updated Model Component of the DG to reflect morphing to component from AB3 to TAPA.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -545,18 +535,17 @@ After which, a new `AssignCommand` object will be created, and is subsequently e
 
    <div markdown="span" class="alert alert-info">:information_source:
    <b>Note:</b> 
-   If no student(s) with a matching `studentId` or `moduleCode`is found, then `PersonNotFoundException()` or `ModuleCodeNotFoundException()` will be thrown.
+   If no student(s) with a matching studentId or moduleCodeis found, then PersonNotFoundException() or ModuleCodeNotFoundException() will be thrown.
```
Comment (brelkh) [inline docs/DeveloperGuide.md:538]:
```suggestion
   If no student(s) with a matching studentId or moduleCode is found, then PersonNotFoundException() or ModuleCodeNotFoundException() will be thrown.
```

**Review by brelkh — APPROVED** [review]
Thanks for updating the model UML too! LGTM, just a minor typo spotted. Also, just wondering since you removed the code snippets, is that to make the DG more readable?

**Comment by Benjykoh** [comment]
> Thanks for updating the model UML too! LGTM, just a minor typo spotted. Also, just wondering since you removed the code snippets, is that to make the DG more readable?

Yea, because I think the ` function wasn't working as intended inside like Note or Warning blocks, and I thought removing it would be less distracting for the reader, do you think there might be a better way to do this?

**Comment by brelkh** [comment]
@Benjykoh Ahhh I see. Admittedly, I think the amount of code snippets in our DG can be quite overwhelming and more difficult to read. That said, the code snippets within warning blocks seem to appear normally for me in the DG online (although they don't appear properly for me in Intellij) :

image

However, if you feel that it can be distracting, then I support removing them too. 👍  Here's a screenshot of what the DG looks like to me after your PR, for reference:
image

**Comment by Benjykoh** [comment]
> @Benjykoh Ahhh I see. Admittedly, I think the amount of code snippets in our DG can be quite overwhelming and more difficult to read. That said, the code snippets within warning blocks seem to appear normally for me in the DG online (although they don't appear properly for me in Intellij) :
> 
> image
> 
> However, if you feel that it can be distracting, then I support removing them too. 👍 Here's a screenshot of what the DG looks like to me after your PR, for reference: image

@brelkh alright, maybe we can take this to the telegram group and ask what the rest think because it was not appearing right for me.

**Pull request merged**